### DISCUSSION TITLE
The Discovery does n't work

### DISCUSSION THREAD

**Discussion by mikekgr** [rustdesk/rustdesk#844] [General]
Dear Sirs,
first of all congratulation for this project. Is VERY good
I am using Self-Host server (Ubuntu 20.04.4), running both hbbr & hbbs and all seems fine using Windows clients to get connected together, but the Discovery doesn't work or I have misunderstood the functionality of this function. I am expecting inside the "Discovery" window to have the registered clients to the server correct?  What to check?

Thanks and Best Regards,
Mike Kranidis

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/pull/337
https://github.com/rustdesk/rustdesk/issues/261

**Reply by mikekgr** [reply]
Dear @rustdesk 
In my case, even inside LAN, the discovery does not seems to work.
Also in case of Self-Host server, is there any way to get the list of online clients ?
Thanks.

**Comment by rustdesk** [comment]
@fufesou  seems still buggy, we may need to refactor it once we start to work on WOL.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/issues/129

**Reply by fufesou** [reply]
Ok.
The discovery also does not work on my PCs.

**Comment by rustdesk** [comment]
Looks work well for him https://github.com/rustdesk/rustdesk/issues/774

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/845

**Comment by fufesou** [comment]
@mikekgr Hi, do you have multiple network adapters? There's a problem on that situation.
I'll fix broadcast with multiple network adapter in next PR.

**Reply by mikekgr** [reply]
Dear @fufesou , no I don't have multiple activated nics, I mostly work on WiFi. As far as I have seen, the reported problem appears on my Windows 10 enterprise edition and not on the other computers on the same lan that are Linux and Windows 10 home. In these computers the lan Discovery works good. Thanks for your fixing tries. Let me know if I can do anything to help you, I will be very glad.

**Reply by fufesou** [reply]
@mikekgr Thank you for your kind help. I've upload temporary builds for windows and linux. Could you please [download](https://github.com/fufesou/rustdesk/releases/tag/tmp) and test lan discovery.  
1. Uninstall your current rustdesk.
2. Run the downloaded builds as user process.
3. Check if lan discovery woks well on you computers.
4. If you need to remove current discovered peers. Delete the peer files. (Win - C:\Users\%Username%\AppData\Roaming\RustDesk\config\RustDesk_lan_peers.toml) (Linux - ~/.config/rustdesk/RustDesk_lan_peers.toml)

**Reply by mikekgr** [reply]
Ok I will be able to make these testings when return home in 4 hours approximately. Then I will report you the results.
Thanks.

**Reply by rustdesk** [reply]
Thanks

**Reply by mikekgr** [reply]
Dear @fufesou ,
I finished my tests using your temporary build. I carefully followed your instructions, but the rusult wasn't better than before,
To explain it better, I still can't see / discover the rest of the stations that are on my LAN and they running the 1.1.9 original version, also tested with this temporary build, I run all three Windows 10 with the temporary build but then, I lost the discovery even on Windows 10 Home that always using the 1.1.9 version was OK the discovery. From these three, the discovery (when running 1.1.9) was problematic only at my Windows 10 Enterprise, at the Windows 10 Home there is not problem at all.
When I uninstall version 1.1.9 and (for all three) I runnig your temporary build, then the discovery stopped for all three independent if they running Windows 10 Home or Enterprise. The conclusion is, don't change your code, doesn't work better on the contrary break the discovery even on Windows 10 Home that they working fine with 1.1.9.
Thanks a lot for your tries.
Best Regards to all the team 

> @mikekgr Thank you for your kind help. I've upload temporary builds for windows and linux. Could you please [download](https://github.com/fufesou/rustdesk/releases/tag/tmp) and test lan discovery.
> 
> 1. Uninstall your current rustdesk.
> 2. Run the downloaded builds as user process.
> 3. Check if lan discovery woks well on you computers.
> 4. If you need to remove current discovered peers. Delete the peer files. (Win - C:\Users%Username%\AppData\Roaming\RustDesk\config\RustDesk_lan_peers.toml) (Linux - ~/.config/rustdesk/RustDesk_lan_peers.toml)

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/issues/261#issuecomment-1179471558 @laeo

**Reply by fufesou** [reply]
Dear @mikekgr  Can you please do the following steps:
1. Delete the discovery file `C:\Users%Username%\AppData\Roaming\RustDesk\config\RustDesk_lan_peers.toml`.
2. Run discovery on your PCs.
3. Check and paste the discovery file here.

Thank you

**Reply by mikekgr** [reply]
> Dear @mikekgr Can you please do the following steps:
> 
> 1. Delete the discovery file `C:\Users%Username%\AppData\Roaming\RustDesk\config\RustDesk_lan_peers.toml`.
> 2. Run discovery on your PCs.
> 3. Check and paste the discovery file here.
> 
> Thank you

I have done this already, the new generated RustDesk_lan_peers.toml just have the below:
peers = '[]'

Tell me if you need anything else, I am glad to help you.

**Reply by fufesou** [reply]
@mikekgr I still want to see the logs:
Please pack the log files in C:\Users%Username%\AppData\Roaming\RustDesk\log and send me an email.
shuanglognchen@yeah.net

**Reply by mikekgr** [reply]
> @mikekgr I still want to see the logs: Please pack the log files in C:\Users%Username%\AppData\Roaming\RustDesk\log and send me an email. [shuanglognchen@yeah.net](mailto:shuanglognchen@yeah.net)

Done ! My email sent with subject: LOG zip file from Mike Kranidis

**Reply by fufesou** [reply]
@mikekgr Sorry for the wrong address.  Please send to shuanglongchen@yeah.net .

**Reply by mikekgr** [reply]
> @mikekgr Sorry for the wrong address. Please send to [shuanglongchen@yeah.net](mailto:shuanglongchen@yeah.net) .

Done, please check
